xmodule/tests/test_library_content.py

```diff
@@ -736,3 +739,130 @@ def test_removed_invalid(self):
+        assert len(all_the_blocks) == 4
+        assert more_blocks < all_the_blocks
+        assert all_the_blocks == {child.block_id for child in self.lc_block.children}
+
+        # Toss a new block into the library and sync.
+        self.make_block("html", self.library, data="new guy in town")
+        self._sync_lc_block_from_library(upgrade_to_latest=True)
+        assert len(self.lc_block.children) == 5
+
+        # Since max_count=-1, it should be added to the selection.
+        all_the_blocks_plus_another = self._select_children()
+        assert len(all_the_blocks_plus_another) == 5
+        assert all_the_blocks < all_the_blocks_plus_another
+        assert all_the_blocks_plus_another == {child.block_id for child in self.lc_block.children}
+
+    def test_overlimit_blocks_removed(self):
+        """
+        Test that decreasing the "max_count" value leads a reduced version of the original subset.
+        """
+        # Start with max
+        self.lc_block.max_count = -1
+        all_the_blocks = self._select_children()
+        assert len(all_the_blocks) == 4
+        assert all_the_blocks == {child.block_id for child in self.lc_block.children}
+
+        # Then drop it down to 3... should be a subset
+        self.lc_block.max_count = 3
+        some_blocks = self._select_children()
+        assert len(some_blocks) == 3
+        assert some_blocks < all_the_blocks
+
+        # Then drop it down to 2... should be a smaller subset
+        self.lc_block.max_count = 2
+        few_blocks = self._select_children()
+        assert len(few_blocks) == 2
+        assert few_blocks < some_blocks
+
+    def test_invalid_blocks_replaced(self):
+        """
+        Test that if a block is deleted from the children (removed from the library),
+        the selected blocks are preserved, except for that now invalid block, which is
+        removed and replaced with a new block at random.
+        """
+        self.lc_block.max_count = 2
+        old_selection = self._select_children()
+        assert len(old_selection) == 2
+
+        keep_block_id, delete_block_id = list(old_selection)
+        self.lc_block.children = [
+            child for child in self.lc_block.children if child.block_id != delete_block_id
+        ]
+
+        new_selection = self._select_children()
+        assert len(old_selection) == 2
+        assert keep_block_id in new_selection
+        assert delete_block_id not in new_selection
+
+
+class TestLibraryContentSelectionInManualMode(LibraryContentTest):
+    """
+    Test the content filtering and selection features which allow specific reference of content.
+    """
+    def setUp(self):
+        super().setUp()
+        self._sync_lc_block_from_library()
+        self.lc_block = self.store.get_item(self.lc_block.location)
+        self.lc_block.shuffle = False
+        self.lc_block.manual = True
+        self._bind_course_block(self.lc_block)
+
+    def test_selects_children_from_candidates(self):
```

Comment:
This test had a few issues, including that its name didn't start with `test_`, so it wasn't running. I've a couple things up, but its assertion is still currently failing.

Reply:
Fixed in https://github.com/openedx/edx-platform/pull/33511/commits/675fed6397ac90cb8ced7269b932651fa4e62cc1